core/src/main/java/de/danoeh/antennapod/core/service/playback/PlaybackService.java

```diff
@@ -1250,21 +1250,23 @@ private void updateMediaSession(final PlayerStatus playerStatus) {
 
         UiModeManager uiModeManager = (UiModeManager) getApplicationContext()
                 .getSystemService(Context.UI_MODE_SERVICE);
-        if (uiModeManager.getCurrentModeType() == Configuration.UI_MODE_TYPE_CAR) {
+        // Hack to work around this bug https://issuetracker.google.com/issues/207389461?pli=1
+        if (Build.VERSION.SDK_INT >= 31
+                || uiModeManager.getCurrentModeType() == Configuration.UI_MODE_TYPE_CAR) {
             sessionState.addCustomAction(
-                new PlaybackStateCompat.CustomAction.Builder(
-                        CUSTOM_ACTION_REWIND,
-                        getString(R.string.rewind_label), R.drawable.ic_notification_fast_rewind)
-                        .build());
+                    new PlaybackStateCompat.CustomAction.Builder(
+                            CUSTOM_ACTION_REWIND,
+                            getString(R.string.rewind_label), R.drawable.ic_notification_fast_rewind)
+                            .build());
             sessionState.addCustomAction(
-                new PlaybackStateCompat.CustomAction.Builder(
-                        CUSTOM_ACTION_FAST_FORWARD,
-                        getString(R.string.fast_forward_label), R.drawable.ic_notification_fast_forward)
-                        .build());
+                    new PlaybackStateCompat.CustomAction.Builder(
+                            CUSTOM_ACTION_FAST_FORWARD,
+                            getString(R.string.fast_forward_label), R.drawable.ic_notification_fast_forward)
+                            .build());
         } else {
             // This would give the PIP of videos a play button
             capabilities = capabilities | PlaybackStateCompat.ACTION_PLAY;
-            if (uiModeManager.getCurrentModeType() == Configuration.UI_MODE_TYPE_WATCH) {
+            if (Build.VERSION.SDK_INT >= 31 || uiModeManager.getCurrentModeType() == Configuration.UI_MODE_TYPE_WATCH) {
```

Comment:
The condition here is never met, isn't it? All API31 devices take the first branch of the surrounding "if" statement and never end up here.

Does the API31 check break the play button of the picture-in-picture videos? The first comment in the "else" branch talks about that and is then no longer called.

Reply:
After posting my last comment, I just checked video podcasts on my device and the PR does break the play button in PIP mode, which would also be why it doesn't add the watch buttons, as you mentioned. See below:

![Video Podcast in Picture in Picture mode does not have a play button](https://user-images.githubusercontent.com/42682253/149461778-9d224f86-0dc2-4beb-aa4c-038c8f45f963.png)